Inddex reports very rough draft

esoergel opened the pull request:
[Doc with a bunch of links](https://docs.google.com/document/d/1fy7_JYhZwy4z8VA8ikygXSXTTSHNqaTFUWaj1AHcVD0/edit)

##### SUMMARY
This is the reports mostly in the state handed over by soldev.  They are as-yet incomplete, but I'd like to merge this in now so Nick can play with them live.  This PR includes:

- The remainder of the work soldev had done
- Misc cleanup of that work undertaken while exploring it
- A number of minor fixes and updates from comparing against the new spec
- Automated tests
- A management command `populate_inddex_test_domain` which bootstraps a domain for local testing

I do plan to rewrite and remove a lot of this in subsequent PRs.

##### FEATURE FLAG
<!--- If this is specific to a feature flag, which one? -->

##### PRODUCT DESCRIPTION
<!--- For non-invisible changes, describe user-facing effects. -->

Commit:
[INDDEX] Add CouchDB management feature

Commit:
[INDDEX] Simplify CouchDB management feature

Commit:
[INDDEX] Add tests for CouchDB management feature

Commit:
[INDDEX] Refactor CouchDB management feature

Commit:
[INDDEX] Adjust filters for drafts

Commit:
[INDDEX] Adjust data providers mixins

Commit:
[INDDEX] Add 'Master Data File' report

Commit:
[INDDEX] Adjust drafts

Commit:
[INDDEX] Update hardcoded data

Commit:
[INDDEX] Add descriptions

kaapstorm committed:
Pass a string not a tuple

kaapstorm committed:
Clean up FixtureDataType._db

kaapstorm committed:
Revert "Clean up FixtureDataType._db"

This reverts commit 7ec3274c4bbe539bde70559b7b23066b64e8906e.

kaapstorm committed:
Revert "Pass a string not a tuple"

This reverts commit abc29860fd29914cd603224b7fbbd0114d5243cb.

kaapstorm committed:
Add view method to mock Couch

Commit:
Add new columns to the data source

Commit:
Adjust reports mixins and add some of CouchDb calculations

Commit:
Add calculations for 'Master Data File'

Commit:
Add calculations for 'Output 2a - Gaps Summary By Food Type'

Commit:
Add calculations for 'Output 2b - Detailed Information on Gaps'

Commit:
Cosmetic changes

esoergel committed:
Add utilities to set up test data

esoergel committed:
Add management command to set up local test domain

esoergel committed:
Rename obnoxiously verbose function

esoergel committed:
remove redundant _in_domain from fn name

esoergel committed:
Painstakingly import fixtures

esoergel committed:
rebuild datasource during import step

esoergel committed:
make strings easier to read

esoergel committed:
fixup to "add fixtures"

esoergel committed:
Test that report headers are correct

esoergel committed:
Get recalled_date from foodrecall case

esoergel committed:
Get age in months and years from the case

esoergel committed:
Add base_term_food_code

esoergel committed:
There should be an underscore in case_id

esoergel committed:
conv_method, conv_option += _code

esoergel committed:
Update several simple variable names

esoergel committed:
Add other_tag_10 where it was missing

esoergel committed:
Add/fix noop column names to match expected report

esoergel committed:
check that rows match

esoergel committed:
Merge in single-use mixin and remove dupl methods

esoergel committed:
These classes aren't mixins, don't call them so

esoergel committed:
These mixins are always used with MasterReportData

esoergel committed:
isort

esoergel committed:
In-line simple function

esoergel committed:
Remove needless indirection

esoergel committed:
The test report filters on hardcoded dates

esoergel committed:
Change name back, fixing broken references

This method used to be called record_to_dict, and there are a bunch of
places which still refer to it as such.  Revert the name change to fix
those references

esoergel committed:
Coerce months to int

esoergel committed:
Do a better check than length comparison

esoergel committed:
Cache fixture lookup, since it's mad slow

esoergel committed:
fixture dbaccessor names

esoergel committed:
Remove unused class

esoergel committed:
Remove total_row that's always None anyways

esoergel committed:
In-line FoodSqlData to simplify

esoergel committed:
Simplify class used to get owner names

esoergel committed:
Rename

esoergel committed:
id_field is never overridden

esoergel committed:
Use separate domain for testing

esoergel committed:
Exclude rows not pulled from cases for now

esoergel committed:
Order rows consistently for easier comparison

esoergel committed:
Compare individual rows

sravfeyn commented on corehq/apps/fixtures/dbaccessors.py:
```diff
@@ -54,6 +54,16 @@ def iter_fixture_items_for_data_type(domain, data_type_id):
         yield FixtureDataItem.wrap(row['doc'])
 
 
+def count_fixture_items(domain, data_type_id):
```
Looks like this is only used in tests, not a big deal to be here, but just a note.

sravfeyn commented on custom/inddex/couch_db_data_collector.py:
```diff
@@ -0,0 +1,105 @@
+
+class CouchDbDataCollector:
+    """
+    This class simplifies getting and filtering data from CouchDB for a specific domain
```
```suggestion
     This class simplifies getting and filtering fixture data from CouchDB for a specific domain
```

esoergel committed:
Update custom/inddex/couch_db_data_collector.py

Co-Authored-By: Sravan Reddy <sravfeyn@gmail.com>

esoergel committed:
Patch call to make tests pass

It seems a bit silly to patch a bunch of stuff so that CouchDbDataCollectorTest
doesn't actually hit CouchDB.  I'm anticipating further changes or removal of
that class though, so I'm postponing any restructuring of its tests.

kaapstorm reviewed (approved):
Definitely non-blocking: `isort`.

At some point in the future, we can think about merging duplicated functionality between `CouchDbDataCollector` and `custom.abt.reports.fixture_utils` ... one day ... if we're gonna need it.
